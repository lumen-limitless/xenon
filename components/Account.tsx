'use client';

import { User2 } from 'lucide-react';
import { Session } from 'next-auth';
import { signIn, signOut } from 'next-auth/react';
import Link from 'next/link';
import GoogleSVG from 'public/svg/google.svg';
import { Avatar, AvatarFallback, AvatarImage } from './ui/avatar';
import { Button } from './ui/button';
import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuGroup,
  DropdownMenuItem,
  DropdownMenuLabel,
  DropdownMenuSeparator,
  DropdownMenuTrigger,
} from './ui/dropdown-menu';
import {
  Sheet,
  SheetContent,
  SheetHeader,
  SheetTitle,
  SheetTrigger,
} from './ui/sheet';

type AccountProps = {
  session: Session | null;
};

export const Account: React.FC<AccountProps> = ({ session }) => {
  return (
    <>
      <Sheet>
        <DropdownMenu modal={false}>
          <DropdownMenuTrigger asChild>
            <Button variant={'ghost'}>
              <Avatar className="h-6 w-6">
                <AvatarImage src={session?.user.image ?? ''} />
                <AvatarFallback>
                  <User2 className="h-6" />
                </AvatarFallback>
              </Avatar>
              <span className="sr-only">Account</span>
            </Button>
          </DropdownMenuTrigger>
          <DropdownMenuContent className="w-56">
            <DropdownMenuLabel>My Account</DropdownMenuLabel>
            <DropdownMenuSeparator />
            {session !== null ? (
              <>
                {' '}
                <DropdownMenuGroup>
                  <DropdownMenuItem asChild>
                    <Link href="/account"> View Account</Link>
                  </DropdownMenuItem>

                  <DropdownMenuItem onClick={() => signOut()}>
                    Sign out
                  </DropdownMenuItem>
                </DropdownMenuGroup>
                {session.user.role === 'ADMIN' && (
                  <>
                    <DropdownMenuSeparator />
                    <DropdownMenuLabel>Administration</DropdownMenuLabel>
                    <DropdownMenuSeparator />
                    <DropdownMenuGroup>
                      <DropdownMenuItem asChild>
                        <Link href="/studio">Content Studio</Link>
                      </DropdownMenuItem>
                      <DropdownMenuItem asChild>
                        <Link href="/manage/products">Store Management</Link>
                      </DropdownMenuItem>
                    </DropdownMenuGroup>
                  </>
                )}
              </>
            ) : (
              <>
                <DropdownMenuGroup>
                  <SheetTrigger asChild>
                    <DropdownMenuItem>Sign in</DropdownMenuItem>
                  </SheetTrigger>
                </DropdownMenuGroup>
              </>
            )}

            <DropdownMenuSeparator />

            <DropdownMenuItem asChild>
              <a
                target="_blank"
                rel="noreferrer"
                href="https://github.com/lumen-limitless/xenon"
              >
                GitHub
              </a>
            </DropdownMenuItem>
            <DropdownMenuItem>Support</DropdownMenuItem>
          </DropdownMenuContent>
        </DropdownMenu>

        <SheetContent className="w-screen">
          <SheetHeader>
            <SheetTitle>Sign In to Xenon</SheetTitle>
          </SheetHeader>
          <div className="flex flex-col items-center justify-center gap-5 py-10">
            <Button
              onClick={() => signIn('google')}
              variant={'outline'}
              size="lg"
            >
              <GoogleSVG className="h-6 w-6" />
              Sign in with Google
            </Button>
          </div>
        </SheetContent>
      </Sheet>
    </>
  );
};
