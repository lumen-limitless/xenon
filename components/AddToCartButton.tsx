'use client'

import { type ButtonProps } from '@/components/ui/button'
import { updateCartAction } from '@/lib/actions'
import { truncateText } from '@/lib/utils'
import { type Product } from '@prisma/client'
import { Loader2 } from 'lucide-react'
import { useRouter } from 'next/navigation'
import { useTransition } from 'react'
import { Button } from './ui/button'
import { toast } from './ui/use-toast'

type AddToCartButtonProps = {
  product: Product
} & Omit<
  ButtonProps,
  'children' | 'onClick' | 'asChild' | 'disabled' | 'aria-disabled'
>

export const AddToCartButton: React.FC<AddToCartButtonProps> = ({
  product,
  ...props
}) => {
  const [isPending, startTransition] = useTransition()
  const router = useRouter()

  return (
    <Button
      {...props}
      aria-disabled={isPending}
      disabled={isPending}
      onClick={() =>
        startTransition(async () => {
          const { success } = await updateCartAction({
            productId: product.id,
            value: 1,
          })
          if (!success) {
            toast({
              title: 'Error',
              description: 'There was an error adding to your bag.',
            })
            return
          }

          toast({
            title: 'Added to bag',
            description: `${truncateText(
              product.title,
              25,
            )} was added to your bag.`,
          })
          router.refresh()
        })
      }
    >
      {isPending ? (
        <Loader2 className="animate-spin" size={24} />
      ) : (
        'Add to bag'
      )}
    </Button>
  )
}
