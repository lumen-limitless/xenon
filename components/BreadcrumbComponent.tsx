'use client';

import { capitalize, cn } from '@/lib/utils';
import { Home } from 'lucide-react';
import Link from 'next/link';
import { usePathname } from 'next/navigation';
import React from 'react';
import {
  Breadcrumb,
  BreadcrumbItem,
  BreadcrumbLink,
  BreadcrumbList,
  BreadcrumbSeparator,
} from './ui/breadcrumb';

type BreadcrumbComponentProps = {
  className?: string;
  homeHref?: string;
};

export const BreadcrumbComponent: React.FC<BreadcrumbComponentProps> = ({
  className,
  homeHref,
}) => {
  const pathname = usePathname();
  const pathParts = pathname.split('/').filter((part) => part !== '');

  return (
    <>
      <Breadcrumb className={cn(className)}>
        <BreadcrumbList>
          <BreadcrumbItem>
            <BreadcrumbLink asChild>
              <Link href={homeHref ?? '/'}>
                <Home size={16} />
                <span className="sr-only">Home</span>
              </Link>
            </BreadcrumbLink>
          </BreadcrumbItem>
          {pathParts.map((part, index) => (
            <React.Fragment key={index}>
              <BreadcrumbSeparator />
              <BreadcrumbItem>
                <BreadcrumbLink asChild>
                  <Link href={`/${pathParts.slice(0, index + 1).join('/')}`}>
                    {capitalize(part).slice(0, 25) +
                      (part.length > 25 ? '...' : '')}
                  </Link>
                </BreadcrumbLink>
              </BreadcrumbItem>
            </React.Fragment>
          ))}
        </BreadcrumbList>
      </Breadcrumb>
    </>
  );
};
