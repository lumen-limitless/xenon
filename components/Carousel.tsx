'use client';

import { AnimatePresence, motion } from 'framer-motion';
import React, { useCallback, useEffect, useState } from 'react';

type CarouselProps = {
  children?: React.ReactNode;
  autoScroll?: boolean;
  controls?: boolean;
};

const swipeConfidenceThreshold = 10000;

export const Carousel: React.FC<CarouselProps> = ({ children, autoScroll }) => {
  const [current, setCurrent] = useState(0);
  const [direction, setDirection] = useState(0);

  const slides = React.Children.toArray(children);

  // Thresholds
  const swipePower = (offset: number, velocity: number) => {
    return Math.abs(offset) * velocity;
  };

  // Swipe handlers
  const paginate = useCallback(
    (newDirection: number) => {
      setCurrent(
        (current) => (current + newDirection + slides.length) % slides.length,
      );
      setDirection(newDirection);
    },
    [slides.length],
  );

  useEffect(() => {
    if (!autoScroll) return;
    const timer = setInterval(() => {
      paginate(1); // go to the next slide
    }, 8000);

    return () => {
      clearInterval(timer); // clear the interval when the component is unmounted
    };
  }, [current, direction, autoScroll, paginate]); // re-run the effect when the `current` or `direction` state changes

  return (
    <AnimatePresence initial={false} custom={direction} mode="wait">
      <motion.div
        className="relative h-full w-full"
        key={current}
        custom={direction}
        initial={{ opacity: 0 }}
        animate={{ opacity: 1 }}
        exit={{ opacity: 0 }}
        transition={{
          ease: 'easeInOut',
          duration: 0.5,
        }}
        drag="x"
        dragConstraints={{ left: 0, right: 0 }}
        dragElastic={1}
        onDragEnd={(e, { offset, velocity }) => {
          const swipe = swipePower(offset.x, velocity.x);

          if (swipe < -swipeConfidenceThreshold) {
            paginate(1);
          } else if (swipe > swipeConfidenceThreshold) {
            paginate(-1);
          }
        }}
      >
        {React.Children.toArray(children)[current]}
      </motion.div>
    </AnimatePresence>
  );
};
