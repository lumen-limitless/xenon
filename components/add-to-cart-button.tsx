'use client';

import { type ButtonProps } from '@/components/ui/button';
import { updateCartAction } from '@/lib/actions';
import { truncateText } from '@/lib/utils';
import type { Product } from '@/types';
import { useMutation } from '@tanstack/react-query';
import { Loader2 } from 'lucide-react';
import { toast } from 'sonner';
import { Button } from './ui/button';

type AddToCartButtonProps = {
  product: Product;
  variantId?: string;
} & Omit<
  ButtonProps,
  'children' | 'onClick' | 'asChild' | 'disabled' | 'aria-disabled'
>;

export const AddToCartButton: React.FC<AddToCartButtonProps> = ({
  product,
  variantId,
  ...props
}) => {
  const { mutate: updateCart, isPending } = useMutation({
    mutationFn: updateCartAction,
  });

  return (
    <Button
      {...props}
      aria-disabled={isPending}
      disabled={isPending}
      onClick={() => {
        updateCart(
          {
            productId: product.id,
            variantId,
            value: 1,
          },
          {
            onError: () => {
              toast.error("Couldn't add to bag. Please try again later.");
            },
            onSuccess: ({ success }) => {
              if (!success) {
                toast.error("Couldn't add to bag. Please try again later.");
                return;
              } else {
                toast.success(
                  `Added ${truncateText(product.title, 20)} to bag`,
                );
              }
            },
          },
        );
      }}
    >
      {isPending ? (
        <Loader2 className="animate-spin" size={24} />
      ) : (
        'Add to bag'
      )}
    </Button>
  );
};
