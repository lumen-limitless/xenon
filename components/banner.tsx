'use client';
import { cn } from '@/lib/utils';
import { MessageSquare } from 'lucide-react';
import { Slideshow } from './slideshow';

type BannerProps = {};

export const Banner: React.FC<BannerProps> = ({}) => {
  return (
    <div
      className={cn(
        'relative flex w-full items-center justify-center bg-foreground px-5 py-2 text-background backdrop-blur',
      )}
    >
      <Slideshow autoScroll>
        <p className="truncate text-nowrap">
          This is an example ecommerce web application
        </p>
        <p className="truncate text-nowrap">
          It is built using Next.js, TailwindCSS, and TypeScript
        </p>
      </Slideshow>

      <a className="hidden items-center gap-1 md:flex" href="tel:111-123-4567">
        <MessageSquare className="h-6 stroke-background" />
        <p className="whitespace-nowrap">Chat or Call (111) 123-4567</p>
      </a>
    </div>
  );
};
