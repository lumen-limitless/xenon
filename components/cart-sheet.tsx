'use client';

import { updateCartAction } from '@/lib/actions';
import { useStore } from '@/lib/store';
import { formatDollars, getCartItemPrice, truncateText } from '@/lib/utils';
import { cartItemTable } from '@/schema';
import type { CartInfo } from '@/types';
import { MinusCircle, PlusCircle, ShoppingBag, Trash2 } from 'lucide-react';
import Image from 'next/image';
import Link from 'next/link';
import { useOptimistic, useTransition } from 'react';
import { CheckoutButton } from './checkout-button';
import { Button } from './ui/button';
import { ScrollArea } from './ui/scroll-area';
import {
  Sheet,
  SheetClose,
  SheetContent,
  SheetFooter,
  SheetHeader,
  SheetTitle,
  SheetTrigger,
} from './ui/sheet';
import {
  Table,
  TableBody,
  TableCaption,
  TableCell,
  TableRow,
} from './ui/table';

type CartSheetProps = {
  cart: CartInfo | null;
};
export const CartSheet: React.FC<CartSheetProps> = ({ cart }) => {
  const cartOpen = useStore((state) => state.cartOpen);
  const toggleCartOpen = useStore((state) => state.toggleCartOpen);
  const [isPending, startTransition] = useTransition();

  return (
    <Sheet open={cartOpen} onOpenChange={toggleCartOpen}>
      <SheetTrigger asChild>
        <Button variant={'ghost'} id="cart-button">
          <ShoppingBag />
          <span className="sr-only">Shopping Bag</span>
        </Button>
      </SheetTrigger>

      <SheetContent side="right" className="w-full pb-32">
        <SheetHeader>
          <SheetTitle>Your Shopping Bag</SheetTitle>
        </SheetHeader>
        <ScrollArea className="h-full pr-3">
          <Table>
            {(!cart || cart.size === 0) && (
              <TableCaption>Your shopping bag is empty.</TableCaption>
            )}

            <TableBody>
              {cart?.items
                .sort((a, b) => {
                  return a.product.title.localeCompare(b.product.title);
                })
                .map((item) => {
                  const cartItemPrice = getCartItemPrice(item);

                  const variant = item.product.variants.find(
                    (v) => v.id === item.variantId,
                  );
                  return (
                    <TableRow key={item.id}>
                      <TableCell className="relative flex w-full items-center gap-5">
                        <Button
                          aria-disabled={isPending}
                          disabled={isPending}
                          aria-label="Remove item from cart"
                          variant={'ghost'}
                          size={'icon'}
                          className="absolute right-0 top-0"
                          onClick={() =>
                            startTransition(async () => {
                              await updateCartAction({
                                productId: item.productId,
                                value: 0 - item.quantity,
                              });
                            })
                          }
                        >
                          <Trash2 className="h-4 w-4" />
                        </Button>

                        <SheetClose asChild>
                          <Link href={`/products/${item.product.slug}`}>
                            <Image
                              className="h-auto w-auto"
                              src={
                                item.product.images?.[0]
                                  ? item.product.images?.[0]
                                  : '/img/placeholder.webp'
                              }
                              alt={item.product.title}
                              height={50}
                              width={50}
                            />
                          </Link>
                        </SheetClose>

                        <div className="mt-5 flex-1 space-y-2">
                          <SheetClose asChild>
                            <Link href={`/products/${item.product.slug}`}>
                              {truncateText(item.product.title, 25)}
                            </Link>
                          </SheetClose>

                          <div id="variant">
                            {variant &&
                              Object.values(variant.attributes).map(
                                (attr, i) => (
                                  <span
                                    className="mx-1 text-sm text-muted-foreground"
                                    key={i}
                                  >
                                    {attr}
                                  </span>
                                ),
                              )}
                          </div>

                          <div className="flex w-full items-center">
                            <CartQuantitySelector item={item} />
                            <span>
                              {formatDollars(cartItemPrice * item.quantity)}
                            </span>
                          </div>
                        </div>
                      </TableCell>
                    </TableRow>
                  );
                })}
            </TableBody>
          </Table>
        </ScrollArea>

        <SheetFooter className="py-5">
          {cart && cart.size > 0 && (
            <SheetClose asChild>
              <CheckoutButton amount={cart.subtotal} />
            </SheetClose>
          )}
        </SheetFooter>
      </SheetContent>
    </Sheet>
  );
};

const CartQuantitySelector: React.FC<{
  item: typeof cartItemTable.$inferSelect;
}> = ({ item }) => {
  const [isPending, startTransition] = useTransition();
  const [optimisticQty, updateOptimisticQty] = useOptimistic(
    item.quantity,
    (currentState, optimisticValue: number) => {
      return currentState + optimisticValue;
    },
  );
  // const router = useRouter();

  const update = (value: number) => {
    startTransition(async () => {
      updateOptimisticQty(value);
      await updateCartAction({
        productId: item.productId,
        value: value,
      });

      // router.refresh();
    });
  };
  return (
    <div className="mr-auto flex w-24 items-center gap-1 text-secondary-foreground">
      <Button
        // aria-disabled={isPending}
        // disabled={isPending}
        aria-label="Remove one item from cart"
        variant={'ghost'}
        size={'icon'}
        onClick={() => update(-1)}
      >
        <MinusCircle />
      </Button>
      <span>{optimisticQty}</span>
      <Button
        // aria-disabled={isPending}
        // disabled={isPending}
        aria-label="Add one item to cart"
        onClick={() => update(+1)}
        variant={'ghost'}
        size={'icon'}
      >
        <PlusCircle />
      </Button>
    </div>
  );
};
