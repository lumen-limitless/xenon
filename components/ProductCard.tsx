'use client';
import { calculatePercentageDifference, cn, formatDollars } from '@/lib/utils';
import type { Product } from '@/types';
import Image from 'next/image';
import Link from 'next/link';
import { AddToCartButton } from './AddToCartButton';
import { Badge } from './ui/badge';
import { Button } from './ui/button';

type ProductCardProps = {
  product?: Product;
  className?: string;
};

export const ProductCard: React.FC<ProductCardProps> = ({
  product,
  className,
}) => {
  if (!product) return null;

  const primaryImage = product.images?.[0] ?? '/img/placeholder.webp';

  return (
    <div
      className={cn('flex h-full flex-col', className)}
      id={product.id}
      data-testid={'product-card'}
    >
      <Link
        href={`/products/${product.slug}`}
        className="rounded-md bg-muted p-5"
      >
        <Image
          className="h-32 w-full object-contain object-center"
          src={primaryImage}
          alt={product.title}
          width={100}
          height={100}
          sizes="(max-width: 768px) 50vw, (max-width: 1200px) 33vw"
        />
      </Link>
      <h2 className="truncate text-left text-sm md:text-base">
        {product.title}
      </h2>
      <div>
        {product.salePrice && (
          <Badge variant={'secondary'}>
            {calculatePercentageDifference(
              product.regularPrice,
              product.salePrice,
            )}
            % OFF
          </Badge>
        )}
      </div>

      <div className="flex gap-1">
        {product.salePrice && (
          <p className="text-primary">{formatDollars(product.salePrice)}</p>
        )}
        <p
          className={cn(
            product.salePrice ? 'text-gray-500 line-through' : 'text-primary',
          )}
        >
          {formatDollars(product.regularPrice)}
        </p>
      </div>

      <div className="mt-auto">
        {product.stock === null || product.stock > 0 ? (
          <AddToCartButton className="w-full" product={product} />
        ) : (
          <Button variant={'secondary'} className="w-full" disabled>
            Out of Stock
          </Button>
        )}
      </div>
    </div>
  );
};
