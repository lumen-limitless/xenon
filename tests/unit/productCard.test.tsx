jest.mock('@/lib/actions', () => ({
  updateCartAction: jest.fn(),
}));

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}));

jest.mock('@tanstack/react-query', () => ({
  useMutation: jest.fn(() => ({
    mutate: jest.fn(),
    isPending: false,
  })),
}));

import { ProductCard } from '@/components/ProductCard';
import type { Product } from '@/types';
import { render } from '@testing-library/react';

describe('ProductCard', () => {
  const product: Partial<Product> = {
    id: '1',
    sku: 'product-1',
    title: 'Product 1',
    slug: 'product-1',
    weight: 10,
    status: 'PUBLISHED',
    regularPrice: 1099,
    salePrice: 999,
    stock: 10,
    images: ['https://example.com/product1.jpg'],
    productDescription: 'Product 1 description',
    shortDescription: 'Product 1 short description',
    note: '',
    metadata: {
      size: 'large',
    },
    createdAt: new Date(),
    updatedAt: new Date(),
    createdBy: '',
    updatedBy: '',
  };

  it('should match snapshot', () => {
    const { asFragment } = render(<ProductCard product={product as Product} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display nothing if product is undefined', () => {
    const { container } = render(<ProductCard product={undefined} />);

    expect(container.firstChild).toBeNull();
  });
});
