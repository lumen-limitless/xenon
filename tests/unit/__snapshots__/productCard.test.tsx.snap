// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col rounded-lg border bg-card text-card-foreground shadow-sm"
  >
    <div
      class="p1 flex flex-grow flex-col pt-0 md:p-2"
    >
      <a
        href="/products/product-1"
      >
        <img
          alt="Product 1"
          class="h-32 w-full object-contain object-center"
          data-nimg="1"
          decoding="async"
          height="100"
          loading="lazy"
          sizes="(max-width: 768px) 50vw, (max-width: 1200px) 33vw"
          src="/_next/image?url=https%3A%2F%2Fexample.com%2Fproduct1.jpg&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fproduct1.jpg&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fexample.com%2Fproduct1.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fexample.com%2Fproduct1.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fexample.com%2Fproduct1.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fexample.com%2Fproduct1.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fexample.com%2Fproduct1.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fexample.com%2Fproduct1.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fexample.com%2Fproduct1.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fexample.com%2Fproduct1.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fexample.com%2Fproduct1.jpg&w=3840&q=75 3840w"
          style="color: transparent;"
          width="100"
        />
      </a>
    </div>
    <div
      class="flex items-center p-1 pt-0 md:p-2 flex-col gap-1"
    >
      <h2
        class="text-left text-sm md:text-base"
      >
        Product 1
      </h2>
      <div
        class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground hover:bg-primary/80 mr-auto"
      >
        $10.99
      </div>
      <button
        aria-disabled="false"
        class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 w-full"
      >
        Add to bag
      </button>
    </div>
  </div>
</DocumentFragment>
`;
